Crash when opening Notification just after sign in

On a slow network:
1. Sign in
2. Open the notifications screen
3. Crash:

```
06-03 09:29:51.200      787-787/org.wordpress.android E/AndroidRuntime﹕ FATAL EXCEPTION: main
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.notifications.NotificationsActivity}: android.view.InflateException: Binary XML file line #11: Error inflating class fragment
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1955)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1980)
            at android.app.ActivityThread.access$600(ActivityThread.java:122)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1146)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:4340)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: android.view.InflateException: Binary XML file line #11: Error inflating class fragment
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:697)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:739)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
            at net.simonvt.menudrawer.MenuDrawer.setContentView(MenuDrawer.java:1369)
            at org.wordpress.android.ui.WPActionBarActivity.createMenuDrawer(WPActionBarActivity.java:200)
            at org.wordpress.android.ui.notifications.NotificationsActivity.onCreate(NotificationsActivity.java:66)
            at android.app.Activity.performCreate(Activity.java:4465)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1049)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1919)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1980)
            at android.app.ActivityThread.access$600(ActivityThread.java:122)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1146)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:4340)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.NullPointerException
            at org.wordpress.android.ui.notifications.NotesAdapter.<init>(NotesAdapter.java:37)
            at org.wordpress.android.ui.notifications.NotificationsListFragment.onViewCreated(NotificationsListFragment.java:52)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:917)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1093)
            at android.support.v4.app.FragmentManagerImpl.addFragment(FragmentManager.java:1195)
            at android.support.v4.app.FragmentActivity.onCreateView(FragmentActivity.java:291)
            at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:669)
            at android.view.LayoutInflater.rInflate(LayoutInflater.java:739)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
            at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
            at net.simonvt.menudrawer.MenuDrawer.setContentView(MenuDrawer.java:1369)
            at org.wordpress.android.ui.WPActionBarActivity.createMenuDrawer(WPActionBarActivity.java:200)
            at org.wordpress.android.ui.notifications.NotificationsActivity.onCreate(NotificationsActivity.java:66)
            at android.app.Activity.performCreate(Activity.java:4465)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1049)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1919)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1980)
            at android.app.ActivityThread.access$600(ActivityThread.java:122)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1146)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:4340)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
            at dalvik.system.NativeStart.main(Native Method)
```

`bucket` is null here: https://github.com/wordpress-mobile/WordPress-Android/blob/develop/src/org/wordpress/android/ui/notifications/NotesAdapter.java#L37

--- a/src/org/wordpress/android/ui/notifications/NotificationsListFragment.java
+++ b/src/org/wordpress/android/ui/notifications/NotificationsListFragment.java
@@ -19,6 +19,7 @@ import org.wordpress.android.R;
 import org.wordpress.android.models.Note;
 import org.wordpress.android.ui.PullToRefreshHelper;
 import org.wordpress.android.util.SimperiumUtils;
+import org.wordpress.android.util.ToastUtils;
 
 import uk.co.senab.actionbarpulltorefresh.library.PullToRefreshLayout;
 
@@ -49,13 +50,17 @@ public class NotificationsListFragment extends ListFragment implements Bucket.Li
 
         // setup the initial notes adapter, starts listening to the bucket
         mBucket = SimperiumUtils.getNotesBucket();
-        mNotesAdapter = new NotesAdapter(getActivity(), mBucket);
 
         ListView listView = getListView();
         listView.setChoiceMode(ListView.CHOICE_MODE_SINGLE);
         listView.setDivider(getResources().getDrawable(R.drawable.list_divider));
         listView.setDividerHeight(1);
-        setListAdapter(mNotesAdapter);
+        if (mBucket != null) {
+            mNotesAdapter = new NotesAdapter(getActivity(), mBucket);
+            setListAdapter(mNotesAdapter);
+        } else {
+            ToastUtils.showToast(getActivity(), R.string.error_refresh_notifications);
+        }
 
         // Set empty text if no notifications
         TextView textview = (TextView) listView.getEmptyView();
@@ -78,21 +83,25 @@ public class NotificationsListFragment extends ListFragment implements Bucket.Li
         refreshNotes();
 
         // start listening to bucket change events
-        mBucket.addListener(this);
+        if (mBucket != null) {
+            mBucket.addListener(this);
+        }
     }
 
     @Override
     public void onPause() {
         // unregister the listener and close the cursor
-        mBucket.removeListener(this);
-
+        if (mBucket != null) {
+            mBucket.removeListener(this);
+        }
         super.onPause();
     }
 
     @Override
     public void onDestroyView() {
-        mNotesAdapter.closeCursor();
-
+        if (mNotesAdapter != null) {
+            mNotesAdapter.closeCursor();
+        }
         mFauxPullToRefreshHelper.unregisterReceiver(getActivity());
         super.onDestroyView();
     }

